File: parking-domain/src/main/java/com/chatchatabc/parking/application/rest/service/JwtService.java
package com.chatchatabc.parking.application.rest.service;

import com.chatchatabc.parking.domain.model.User;

public interface JwtService {

    /**
     * Generate a JWT token for the given user id
     *
     * @param userId the user id
     * @return the JWT token
     */
    String generateToken(String userId);

    /**
     * Validate the given token and return the user
     *
     * @param token the token
     * @return the user
     */
    User validateTokenAndGetUser(String token);
}

File: parking-api/src/main/kotlin/com/chatchatabc/parking/api/application/rest/service/JwtService.kt
package com.chatchatabc.parking.api.application.rest.service

import com.chatchatabc.parking.domain.model.User

interface JwtService {

    /**
     * Generate a JWT token for the given user id
     */
    fun generateToken(userId: String): String

    /**
     * Validate the given token and return the user
     */
    fun validateTokenAndGetUser(token: String): User
}
